src/java.base/share/classes/java/lang/reflect/code/type/ClassType.java

```diff
@@ -107,7 +107,7 @@ public String toString() {
                 type.displayName() :
                 type.displayName().substring(enclosing.type.displayName().length() + 1);
         String typeArgs = hasTypeArguments() ?
-                typeArguments().stream().map(JavaType::toString)
+                typeArguments().stream().map(t -> t.externalize().toString())
```

Comment:
Hmm... this change seems suspect. I think we need to update `TestReferences` to test round tripping to an externalized string. `t.toString()` and `t.externalize().toString()`have diverged. With the former it is not generally possible to round trip between the type elements and there strings.